src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecutionTest.java

```diff
@@ -206,6 +211,82 @@ public void testSqlServerBulkCopyCachingConnectionLevel() throws Exception {
         }
     }
 
+    @Test
+    public void testSqlServerBulkCopyCachingConnectionLevelMultiThreaded() throws Exception {
+        Calendar gmtCal = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
+        long ms = 1578743412000L;
+        long timeOut = 90000; // 90 seconds
+        int NUMBER_SIMULTANEOUS_INSERTS = 40;
+
+        try (SQLServerConnection con = (SQLServerConnection) DriverManager.getConnection(connectionString
+                + ";useBulkCopyForBatchInsert=true;cacheBulkCopyMetadata=true;sendTemporalDataTypesAsStringForBulkCopy=false;");
+                Statement stmt = con.createStatement()) {
+
+            // Needs to be on a JDK version greater than 8
+            assumeTrue(TestUtils.getJVMVersion() > 8);
```

Comment:
This is not necessary.
As discussed pls move this line in the other test to beginning of the test and add comment

Reply:
I've moved it above, but it is necessary as we need reflection to clear the cache in case of deadlock.